vignettes/geocoding_simple_output.Rmd

```diff
@@ -0,0 +1,350 @@
+
+census_vintage: a dataset based on the census
+
+
+We only have 4995 observations. So, we will not run parallel processing and set parallel to FALSE in the run_geocoder() function.
+
+
+
+*If there are any missing geocoded addresses, the run_geocoder() function will re-run the geocoder on those missing geocoded addresses. Then, join the missing data back in with the original voter file.*
+
+```{r}
+geocoded_data <- run_geocoder(
+  voter_file = ga_5k,
+  geocoder = "census",
+  parallel = FALSE,
+  voter_id = "registration_number",
+  street = "street_address",
+  city = "city",
+  state = "state",
+  zipcode = "zipcode",
+  country = "US",
+  census_return = "locations",
+  census_benchmark = "Public_AR_Current",
+  census_output = "simple",
+  census_class = "sf",
+  census_vintage = NULL,
+  opencage_key = NULL
+)
+```
+
+Check the column names of the geocoded dataset. There should be an additional column called `geometry` with latitude, longitude coordinates.
+```{r}
+
+# Check the column names of the geocoded_dataset object
+names(geocoded_data)
+```
+
+```{r}
+# Check the first six rows of the geocoded_dataset object
+head(geocoded_data)
+```
+
+```{r}
+# The number of rows missing in new dataframe
+num_miss_geo <- nrow(ga_5k) - nrow(geocoded_data)
+
+# Only re-run the geocoder if missing data is present.
+if (num_miss_geo > 0) {
+
+  # Find non-geocoded data
+  missing_lonlat_df <- anti_join(ga_5k, as.data.frame(geocoded_data))
+
+  # Run the geocoder on the missing data
+  rerun_data <- run_geocoder(
+    voter_file = missing_lonlat_df,
+    geocoder = "census",
+    parallel = FALSE,
+    voter_id = "registration_number",
+    street = "street_address",
+    city = "city",
+    state = "state",
+    zipcode = "zipcode",
+    country = "US",
+    census_return = "locations",
+    census_benchmark = "Public_AR_Current",
+    census_output = "simple",
+    census_vintage = NULL,
+    census_class = "sf",
+    opencage_key = NULL
+  )
+}
+```
+
+*Combine the newly geocoded data from the `rerun_data` object and the original geocoded data, `geocoded_data`, object.*
+```{r}
+geo_combined <- rbind(geocoded_data, rerun_data)
+```
+
+```{r}
+# Check the number of observations in the combined geocoded voter registration dataset
+nrow(geo_combined)
+```
+
+```{r}
+# Check the dimensions of the combined geocoded voter registration dataset
+dim(geo_combined)
+```
+```{r}
+# rename columns
+names(geo_combined)[names(geo_combined) == "cxy_state_id"] <- "STATEFP10"
+names(geo_combined)[names(geo_combined) == "cxy_county_id"] <- "COUNTYFP10"
+names(geo_combined)[names(geo_combined) == "cxy_tract_id"] <- "TRACTCE10"
+names(geo_combined)[names(geo_combined) == "cxy_block_id"] <- "BLOCKCE10"
+```
+
+### Step 6: Save your geocoded file 
+```{r}
+write_csv(geo_combined, "~/shared/BISG_datasets/gwin_fulton_5k_file.csv")
```

Comment:
This line won't be necessary in a vignette - can just view the top of the final dataframe instead.

Reply:
Or just jump to the plots in the next step.